homeassistant/components/swiss_public_transport/sensor.py

```diff
@@ -39,44 +43,123 @@
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
-        vol.Required(CONF_DESTINATION): cv.string,
-        vol.Required(CONF_START): cv.string,
+        vol.Optional(CONF_DESTINATION): cv.string,
+        vol.Optional(CONF_START): cv.string,
+        vol.Optional(CONF_STATIONBOARD): vol.All(cv.ensure_list, [cv.string]),
+        vol.Optional(CONF_LIMIT): cv.positive_int,
         vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
     }
 )
 
 
+async def test_opendata(opendata):
+    """Test wheather the current configuration allows to get data."""
+
+    try:
+        data = await opendata.async_get_data()
+        _LOGGER.debug(data)
+    except OpendataTransportError:
+        _LOGGER.error(
+            "Check at http://transport.opendata.ch/examples/stationboard.html "
+            "if your station names are valid"
+        )
+        return False
+    return True
+
+
 async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Set up the Swiss public transport sensor."""
 
     name = config.get(CONF_NAME)
     start = config.get(CONF_START)
     destination = config.get(CONF_DESTINATION)
+    stationboard = config.get(CONF_STATIONBOARD)
+    limit = config.get(CONF_LIMIT, 5)
```

Comment:
The default should be set in the validation -> `vol.Optional(CONF_LIMIT, default=5): cv.positive_int,`